### PR TITLE
Added a requirements section to README.

### PR DESCRIPTION
**Pull request by andreas-kupries**
## Description

Extended the toplevel README with a section on requirements.

## Motivation and Context

See https://github.com/SUSE/kubecf/issues/295

## How Has This Been Tested?

Documentation changes. Ran them through pandoc to check syntax. (Learned that pandoc's PDF backend (or the latex) does not like the embedded SVG image at the top).

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code has security implications.
- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,6 +10,64 @@ under [Project Quarks].
 [Cloud Foundry Operator]: https://github.com/cloudfoundry-incubator/cf-operator/
 [Project Quarks]:         https://www.cloudfoundry.org/project-quarks/
 
+## Contributing to Kubecf development
```
Comment (f0rmiga) [inline README.md]:
```suggestion
## Contributing to KubeCF development
```

File: README.md [hunk 2]
```diff
@@ -10,6 +10,64 @@ under [Project Quarks].
...
+
+See the [Guide to Contribution](doc/Contribute.md).
+
+# System requirements
+
+To work with `kubecf` a variety of supporting tools are required. These are
```
Comment (f0rmiga) [inline README.md]:
```suggestion
To work with `kubecf`, a variety of supporting tools are required. These are:
```

File: README.md [hunk 3]
```diff
@@ -10,6 +10,64 @@ under [Project Quarks].
...
+
+| Tool		| Notes					|
+|---		|---					|
```
Comment (f0rmiga) [inline README.md]: Could you align the rightmost border, please?

File: README.md [hunk 4]
```diff
@@ -10,6 +10,64 @@ under [Project Quarks].
...
+|k8s		| Foundation.				|
```
Comment (f0rmiga) [inline README.md]:
```suggestion
|Kubernetes	| The platform to run KubeCF.				|
```

File: README.md [hunk 5]
```diff
@@ -10,6 +10,64 @@ under [Project Quarks].
...
+|minikube	| Provider for local k8s clusters.	|
+|kind		| Provider for local k8s clusters.	|
+|kubectl	| Client to talk to k8s clusters.	|
+|Helm (client)	| Handling helm charts.			|
+|Helm (server)	| __Tiller is not required__.		|
```
Comment (f0rmiga) [inline README.md]:
```suggestion
|Helm Tiller	| Necessary with Helm 2.x.		|
```

File: README.md [hunk 6]
```diff
@@ -10,6 +10,64 @@ under [Project Quarks].
...
+|CF Operator	| Processes BOSH deployments. Maps them to kube objects.	|
```
Comment (f0rmiga) [inline README.md]:
```suggestion
|cf-operator	| Processes BOSH deployments. Maps them to kube objects.	|
```

File: README.md [hunk 7]
```diff
@@ -10,6 +10,64 @@ under [Project Quarks].
...
+|cf-deployment	| The CF release at the core of kubecf				|
+
+As most of the developers use the `Bazel` build system coming with the
+`kubecf` repository they implicitly use the versions for the tools set
```
Comment (f0rmiga) [inline README.md]:
```suggestion
`kubecf` repository, they implicitly use the versions for the tools set
```

File: README.md [hunk 8]
```diff
@@ -10,6 +10,64 @@ under [Project Quarks].
...
+down in `Bazel`'s main configuration file, `def.bzl`.
```
Comment (f0rmiga) [inline README.md]:
```suggestion
down in `Bazel`'s main project configuration file, `def.bzl`.
```

File: README.md [hunk 9]
```diff
@@ -10,6 +10,64 @@ under [Project Quarks].
...
+
+| Tool		| Version		|
+|---		|---			|
+|k8s		| 1.15.6		|
+|minikube	| 1.6.2			|
+|kind		| 0.6.0			|
+|kubectl	| 1.15.6		|
+|Helm (client)	| 2.16.1		|
+|Helm (server)	| n/a			|
+|CF Operator	| 2.0.0-0.g0142d1e9	|
```
Comment (f0rmiga) [inline README.md]:
```suggestion
|cf-operator	| 2.0.0-0.g0142d1e9	|
```

File: README.md [hunk 10]
```diff
@@ -10,6 +10,64 @@ under [Project Quarks].
...
+|cf-deployment	| 12.18.0
```
Comment (f0rmiga) [inline README.md]:
```suggestion
|cf-deployment	| 12.18.0		|
```

File: README.md [hunk 11]
```diff
@@ -10,6 +10,64 @@ under [Project Quarks].
...
+
+__Always__ check the contents of `def.bzl` for the authoritative
+answer.
+
+__Note further__: Just because the build system provides targets to
+conveniently bring up a local k8s cluster using minikube or kind, this
+does not preclude the use of other k8s cluster providers, local or in
+the cloud.
+
+Beside the required tools noted above other tools used by developers are
```
Comment (f0rmiga) [inline README.md]:
```suggestion
Besides the required tools noted above, other tools used by developers are:
```

File: README.md [hunk 12]
```diff
@@ -10,6 +10,64 @@ under [Project Quarks].
...
+
+| Tool  | Notes				| Location				|
+|---	|---				|---					|
+|k9s	| Curses-based UI over kubectl	| https://github.com/derailed/k9s	|
+|stern	| Multi-pod log tailing		| https://github.com/wercker/stern	|
+
+Last, but not least __more documentation__ on how to work with
```
Comment (f0rmiga) [inline README.md]:
```suggestion
Last, but not least, __more documentation__ on how to work with
```

File: README.md (hunk 1 above)
Comment (andreas-kupries) [reply README.md]: Sure. Not part of my changes technically, but trivial to fix.
Will fix tabs also.

File: README.md (hunk 2 above)
Comment (andreas-kupries) [reply README.md]: will fix

File: README.md (hunk 3 above)
Comment (andreas-kupries) [reply README.md]: will do

File: README.md (hunk 4 above)
Comment (andreas-kupries) [reply README.md]: will do

File: README.md (hunk 5 above)
Comment (andreas-kupries) [reply README.md]: I believe that this change would be in conflict with your

> I'm saying that Tiller is not needed for kubecf itself. With kubecf we use the helm template command and pipe the result to kubectl apply.

in the jira ticket.

File: README.md (hunk 6 above)
Comment (andreas-kupries) [reply README.md]: ok

File: README.md (hunk 7 above)
Comment (andreas-kupries) [reply README.md]: ok

File: README.md (hunk 8 above)
Comment (andreas-kupries) [reply README.md]: ok

File: README.md (hunk 9 above)
Comment (andreas-kupries) [reply README.md]: ok

File: README.md (hunk 10 above)
Comment (andreas-kupries) [reply README.md]: ok

File: README.md (hunk 11 above)
Comment (andreas-kupries) [reply README.md]: ok

File: README.md (hunk 12 above)
Comment (andreas-kupries) [reply README.md]: ok

File: README.md (hunk 1 above)
Comment (andreas-kupries) [reply README.md]: Addressed with commit 6a472a1a7d9b0ad31af96b14208e31f3d4c9a5cd.

File: README.md (hunk 2 above)
Comment (andreas-kupries) [reply README.md]: Addressed with commit 6a472a1a7d9b0ad31af96b14208e31f3d4c9a5cd.

File: README.md (hunk 3 above)
Comment (andreas-kupries) [reply README.md]: Addressed with commit 6a472a1a7d9b0ad31af96b14208e31f3d4c9a5cd.

File: README.md (hunk 4 above)
Comment (andreas-kupries) [reply README.md]: Addressed with commit 6a472a1a7d9b0ad31af96b14208e31f3d4c9a5cd.

File: README.md (hunk 6 above)
Comment (andreas-kupries) [reply README.md]: Addressed with commit 6a472a1a7d9b0ad31af96b14208e31f3d4c9a5cd.

File: README.md (hunk 7 above)
Comment (andreas-kupries) [reply README.md]: Addressed with commit 6a472a1a7d9b0ad31af96b14208e31f3d4c9a5cd.

File: README.md (hunk 8 above)
Comment (andreas-kupries) [reply README.md]: Addressed with commit 6a472a1a7d9b0ad31af96b14208e31f3d4c9a5cd.

File: README.md (hunk 9 above)
Comment (andreas-kupries) [reply README.md]: Addressed with commit 6a472a1a7d9b0ad31af96b14208e31f3d4c9a5cd.

File: README.md (hunk 10 above)
Comment (andreas-kupries) [reply README.md]: Addressed with commit 6a472a1a7d9b0ad31af96b14208e31f3d4c9a5cd.

File: README.md (hunk 11 above)
Comment (andreas-kupries) [reply README.md]: Addressed with commit 6a472a1a7d9b0ad31af96b14208e31f3d4c9a5cd.

File: README.md (hunk 12 above)
Comment (andreas-kupries) [reply README.md]: Addressed with commit 6a472a1a7d9b0ad31af96b14208e31f3d4c9a5cd.

File: README.md (hunk 5 above)
Comment (andreas-kupries) [reply README.md]: Might be better to say `Necessary with Helm 2.x. when using external charts`.

Comment (f0rmiga) [reply README.md]: OK, I'll try to be more clear. Tiller is the "deployer" of Helm 2.x. It is the server-side component that talks to the Kubernetes API. It's the standard way of deploying when using Helm 2.x.

The confusion is around "Helm using KubeCF" vs "KubeCF using Helm".

It's certainly Helm that uses the KubeCF chart to deploy it, not the other way around. When I say that `kubecf uses the helm template command`, I mean that the scripts under the kubecf repo use the `helm template` command to deploy KubeCF.

People familiar with the process of deploying a Helm chart using Helm 2.x will certainly have Tiller installed.

My bad on not being clear in the Jira ticket. Sorry.

Comment (andreas-kupries) [reply README.md]: Hopefully addressed by commit d256a8546bac8d48a164f62afd1b065ce7cc7ebf

**Pull request merged**